config/cache/v1alpha1/cache.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package gingersnap.config.cache.v1alpha1;
+import "apimachinery/pkg/api/resource/quantity.proto";
+
+// TODO  option java_package = define package naming for java
+// TODO  option java_outer_classname = define outer class name
+option java_multiple_files = true;
+// TODO option go_package = verify below value is ok
+option go_package = "api/v1alpha1";
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes the desired configuration for a Cache. Only DB Cache Service is supported atm
+message CacheSpec {
+  // Resource profile for the cache provider
+  ResourcesProfile resources = 1;
+  // DatasourceRef or a ServiceBindingRef (TODO clarify)
+  NamespacedRef dataSourceRef = 2;
```

Comment:
I think this needs to be `DataSourceSpec`

Reply:
maybe I remeber badly, didn't we say we want a DataSource CR that can be reused?

Reply:
No, then yes, then no, then ... :laughing:

I think if we're referencing a ServiceBinding CR, then having the `DataSourceSpec` as part of the Cache CR is probably fine for simplicity (1 less controller). We can extract this into a separate CR later if it's required.